src/interface/variable_pack.hpp

```diff
@@ -162,9 +212,33 @@ VariableFluxPack<T> MakeFluxPack(const vpack_types::VarList<T> &vars,
           std::pair<std::string, IndexPair>(v->label(), IndexPair(vstart, vindex - 1)));
     }
   }
+  if (!(sparse_name == "")) {
+    vmap->insert(std::pair<std::string, IndexPair>(sparse_name,
+                                                   IndexPair(sparse_start, vindex - 1)));
+  }
   // add fluxes to host view
   vindex = 0;
+  sparse_name = "";
   for (const auto &v : flux_vars) {
+    if (v->IsSet(Metadata::Sparse)) {
+      std::string sparse_trim = v->label();
+      sparse_trim.erase(sparse_trim.find_last_of("_"));
+      if (sparse_name == "") {
+        sparse_name = sparse_trim;
+        sparse_start = vindex;
+      }
+      if (sparse_name != sparse_trim) {
+        vmap->insert(std::pair<std::string, IndexPair>(
+            sparse_name, IndexPair(sparse_start, vindex - 1)));
+        sparse_name = sparse_trim;
+        sparse_start = vindex;
+      }
+
+    } else if (!(sparse_name == "")) {
+      vmap->insert(std::pair<std::string, IndexPair>(
+          sparse_name, IndexPair(sparse_start, vindex - 1)));
+      sparse_name = "";
+    }
```

Comment:
The MakeFluxPack function is really large, I wish it could be broken up into subfunctions which would make it more readable.

Reply:
broke it up.